src/conv-core/isomalloc.C

```diff
@@ -1025,6 +1029,37 @@ struct isommap
     allocated_extent = start;
   }
 
+  void * permanent_alloc(size_t size)
+  {
+    const size_t realsize = CMIALIGN(size, pagesize);
+
+    CmiLock(lock);
+
+    void * const mapped = map_global_memory(allocated_extent, realsize);
+    if (mapped != nullptr)
+      allocated_extent += realsize;
+
+    CmiUnlock(lock);
+    return mapped;
+  }
+
+  void * permanent_alloc(size_t size, size_t align)
+  {
+    const size_t realsize = CMIALIGN(size, pagesize);
+
+    CmiLock(lock);
```

Comment:
Here and above, can we just lock around the call to map_global_memory? We always want to keep critical sections as short as possible when using locks. And if so, should the locks be moved inside map_global_memory?

Reply:
The locks are here to provide exclusive access to the contents of the isommap struct, particularly the allocated_extent member. In the general case the locks should be uncontended because each AMPI rank has its own context. They are only here for the case of something like AMPI+OpenMP.

Reply:
Can you add this as a comment?

Reply:
> Can you add this as a comment?

Added to the definition of the lock in `struct isommap`.